src/core/stream_recv.c

```diff
@@ -162,6 +162,79 @@ QuicStreamRecvQueueFlush(
     }
 }
 
+//
+// Processes a received RELIABLE_RESET frame's payload.
+//
+_IRQL_requires_max_(PASSIVE_LEVEL)
+void
+QuicStreamProcessReliableResetFrame(
+    _In_ QUIC_STREAM* Stream,
+    _In_ QUIC_VAR_INT ErrorCode,
+    _In_ QUIC_VAR_INT ReliableOffset
+    )
+{
+    if (!Stream->Connection->State.ReliableResetStreamNegotiated) {
+        //
+        // The peer tried to use an exprimental feature without
+        // negotiating first. Kill the connection.
+        //
+        QuicTraceLogStreamWarning(
+            ReliableResetNotNegotiated,
+            Stream,
+            "Tried to use ReliableReset without negotiation.");
+        QuicConnTransportError(Stream->Connection, QUIC_ERROR_STREAM_STATE_ERROR);
+        return;
+    }
+
+    if (Stream->RecvMaxLength == 0 || ReliableOffset < Stream->RecvMaxLength) {
+        //
+        // As outlined in the spec, if we receive multiple CLOSE_STREAM frames, we only accept strictly
+        // decreasing offsets.
+        //
+        Stream->RecvMaxLength = ReliableOffset;
+        Stream->Flags.RemoteCloseResetReliable = TRUE;
+
+        QuicTraceLogStreamInfo(
+            ReliableRecvOffsetSet,
+            Stream,
+            "Reliable recv offset set to %llu",
+            ReliableOffset);
+    }
+
+    if (Stream->RecvBuffer.BaseOffset >= Stream->RecvMaxLength) {
+        Stream->Flags.ReceiveClosedReliable = TRUE;
+        Stream->Flags.LocalCloseAcked = TRUE;
+        Stream->Flags.RemoteCloseAcked = TRUE;
+        QuicTraceLogStreamVerbose(
+            StreamProcessReliableReset,
+            Stream,
+            "Shutting down stream from Process Reliable Reset Frame. Recv Side.");
+        QUIC_STREAM_EVENT Event;
+        Event.Type = QUIC_STREAM_EVENT_PEER_SEND_SHUTDOWN;
+        QuicTraceLogStreamVerbose(
+            IndicatePeerSendShutdown,
+            Stream,
+            "Indicating QUIC_STREAM_EVENT_PEER_SEND_SHUTDOWN");
+        (void)QuicStreamIndicateEvent(Stream, &Event);
+        QuicStreamRecvShutdown(Stream, TRUE, QUIC_ERROR_NO_ERROR);
```

Comment:
Why are you passing `QUIC_ERROR_NO_ERROR` here when there is an error being passed in? Also, why are you setting `TRUE` for silent?

Reply:
The prior codepath did exactly the same thing as calling RecvShutdown silently. I can update the error code, though.